content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,221 @@
+
+h2. Processing the message in the Worker
+
+Ably Reactor is monitoring the channel and calls the Cloudflare Worker with the message contents.
+We can see how this works by taking a look at the Worker file again "./cloudflare-worker/index.js":https://github.com/ably/tutorials/blob/reactor-event-cloudflare-javascript/cloudflare-worker/index.js
+
+```[javascript]
+async function publishToAbly(data) {
+    try {
+        const URL = `https://rest.ably.io/channels/cloudflare-bot/messages?key=${key}`;
+        
+        await fetch(`${URL}`, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ data });
+        });
+
+    } catch (error) {
+        return error;
+    }
+}
+```
+The Worker is re-publishing our initial message to a channel called @cloudflare-bot@ using the "Ably REST API":https://www.ably.io/documentation/rest. "We would usually recommend":https://support.ably.io/a/solutions/articles/3000061848 that you use the "REST Client Library API":https://www.ably.io/documentation/rest rather than make a direct POST to the Ably Rest API, since it is more discoverable and easier to use, however in this particular case, installing the library files in the Worker would mean bundling the entire library with your Worker, which would increase the size of your script unnecessarily. 
+
+We'll subscribe to this channel on our dashboard to receieve these notifications.
+
+h2. Connecting the dashboard
+
+In our game app, there's a React Component called Scoreboard in *./src/feature/scoreboard/index.js* which contains the following code on line 12
+
+```[javascript]
+const ably = new Realtime(ABLY_API_KEY);
+
+ably.connection.on('connected', () => {
+    console.log('Successfully connected!');
+    
+    const inboundChannel = ably.channels.get('cloudflare-bot');
+    inboundChannel.subscribe(message => {
+        this.scoreboard.push(message.data.playerSteps);
+        this.setState(this.scoreboard);
+    })
+})
+```
+
+This is similar to the code we used to originally publish the message, but this time, we're subscribing to the channel  @cloudflare-bot@.
+When the Cloudflare Worker re-publishes our initial message onto this channel, the code in our subscription fires.
+
+This is a stateful React component which keeps track of the moves - "playerSteps" that our players make when they finish each level.
+When a message appears on the channel, we're updating our scoreboard array (defined when the component is created), and updating the React-managed state.
+
+As our state is modified, React will re-draw our component with the updated move counts displayed on the page.
+
+Congratulations, you're done!
+
+<a href="/images/tutorials/reactor/reactor-events-cloudflare-gameplay.gif" target="_blank">
+  <img src="/images/tutorials/reactor/reactor-events-cloudflare-gameplay.gif" style="width: 100%" alt="Animated image of the browser app in action">
+</a>
+
+h2. Next Steps
```

Comment:
Worth linking again here to the code repo in /tutorials and also to cloudflare worker docs if they'd like to learn more